app/main.py

```diff
@@ -1,3 +1,50 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, distance: int | float) -> None:
+        self.km = distance
+
+    @staticmethod
+    def is_int(other: int | float | Distance) -> bool:
```

Comment:
seems like is_int name is not good enough, you're also checking for float.

Reply:
it's local function, implementing for checking if value is in int | float, yes, but overall idea is not_distance, 
while is_int_or_float and not_distance is either 2 long, or unnecessary for method name, while is_int is pretty intuitive.
I agree u could make even a little bit more descriptive, but I'm sure it's intuitive enough for understanding, what method does, so further refactoring is unnecessary, as for me.

Reply:
your code should be clear rather than intuitive

Reply:
why don't you want to name it `is_distance` for example, is it bad for you?

Reply:
> why don't you want to name it `is_distance` for example, is it bad for you?

Yup it's great, similar implementation, more intuitive, more clear, idk, sometimes production matters, but for better practice I totally agree